### ISSUE TITLE
builtin: glob

### ISSUE THREAD

**Issue by katcipis** [NeowayLabs/nash#183]
I was thinking about the expanding string thing. I agree with not doing implicitly but it really is annoying when I want to remove all the pdf's from a directory to not have something easy to use.

If we implement a glob we could do something like:

```
a <= glob("*.pdf")
rm -rf $a
```

I always wanted some syntax to express this too:

```
rm -rf glob("*.pdf")
```

Pretty small and more explicit :-)

**Comment by tiago4orion** [comment]
This `glob` function doesn't need to be built in.

```
# we can improve to use wildcard also
fn glob(pat) {
    content <= ls | grep $pat
    files <= split($content, "\n")
    return $files
}

files <= glob(".*\.pdf$")
rm $files
```

About executing function in command arguments, I have an old proposal for that: https://github.com/NeowayLabs/nash/issues/82

My use case was exactly yours.

**Comment by vitorarins** [comment]
I like to see these simple functions (that we use all the time) as a standard in Nash.
IMO if those functions don't come builtin, at least it should be easy to get them as a package during installation or something like that.

**Comment by katcipis** [comment]
I thought that using Go's glob implementation would be a good way to guarantee a non exponential glob, instead of depending on binaries installed on the host, but it may be a crazy trip :-) (excited reading the Russ Cox article).

**Comment by tiago4orion** [comment]
Closed by #209
